File: C++/26.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        int index = 0;
        if (nums.size() == 0)
            return 0;
        for (int i = 0; i < nums.size(); i++){
            if (nums[index] != nums[i]){
                nums[index+1] = nums[i];
                index++;
            }
        }
        return index + 1;
    }
};

int main(int argc, char const *argv[])
{
    Solution s = Solution();
    vector<int> num = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
    int lens = s.removeDuplicates(num);
    for (int i = 0; i < lens; i++){
        cout << num[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Python/26.py
class Solution:
    def removeDuplicates(self, nums):
        if nums == []:
            return 0
        index = 0
        for i in range(1, len(nums)):
            if nums[index] == nums[i]:
                continue
            else:
                nums[index+1] = nums[i]
                index+= 1
        return index + 1


if __name__ == '__main__':
    s = Solution()
    nums = [0, 0, 1, 1, 1, 2, 2, 3, 4, 5, 6, 7]
    lens = s.removeDuplicates(nums)
    print(lens)
    for i in range(lens):
        print(nums[i])